common/theme.c

```diff
@@ -123,3 +134,48 @@ void themeStartup(ThemePreset preset) {
     }
     config_destroy(&cfg);
 }
+
+void GetThemePathFromConfig(const char* themePath) {
+    config_t cfg = {0};
+    config_setting_t *setting;
+    config_init(&cfg);
+
+    char tmp_path[PATH_MAX] = {0};
+
+    #ifdef __SWITCH__
+    tmp_path[0] = '/';
+    #endif
+
+    strncat(tmp_path, "config/nx-hbmenu/setting.cfg", sizeof(tmp_path)-2);
+    bool good_cfg = config_read_file(&cfg, tmp_path);
+    
+    if(good_cfg) {
+        setting = config_lookup(&cfg, "themePath");
+        config_setting_lookup_string(setting, "themePath", &themePath);
+    }
+    config_destroy(&cfg);
+}
+
+void SetThemePathToConfig(const char* themePath) {
+    config_t cfg = {0};
+    config_init(&cfg);
+
+    char settingPath[PATH_MAX] = {0};
+    config_setting_t *root,*group, *setting;
+
+    #ifdef __SWITCH__
+    settingPath[0] = '/';
+    #endif
+
+    strncat(settingPath, "config/nx-hbmenu/setting.cfg", sizeof(settingPath)-2);
+    root = config_root_setting(&cfg);
+    group = config_setting_add(root, "hbmenuConfig", CONFIG_TYPE_GROUP);
```

Comment:
nullptr checks missing.

Reply:
is this needed on every single one? considering i am not reading from a file, but instead creating instances using libconfig functions, would i just need this on the root one?

Reply:
all